code/game/gamemodes/scoreboard.dm

```diff
@@ -149,14 +149,22 @@ GLOBAL_VAR(scoreboard) // Variable to save the scoreboard string once it's been
 		richest_cash = cash_score
 		richest_name = H.real_name
 		richest_job = H.job
-		richest_key = H.key
+		if(H)
+			if(H.client.prefs.toggles2 & PREFTOGGLE_2_ANON)
+				richest_key = "Anon"
+			else
+				richest_key = H.key
```

Comment:
A few things
First of all, you shouldn't need the if(H) check, since we've already done the ishuman(M) check, which ensures it's a human and not null.
Second, client isn't always guaranteed on something with a mind, so it's good to do a nullcheck here.
We can also quickly pull this into a quick line here:
```suggestion
		richest_key = (H.client?.prefs.toggles2 & PREFTOGGLE_2_ANON) ? "Anon" : H.key
```